src/modules/item/sagas.ts

```diff
@@ -153,16 +156,43 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
     }
   }
 
+  function* fetchCollectionItemsWithBatch(collectionId: string, pagesToFetch: number[], limit?: number, isReviewing = false) {
+    const REQUEST_BATCH_SIZE = 10
+    const queue = new PQueue({ concurrency: REQUEST_BATCH_SIZE })
+    const promisesOfPagesToFetch: (() => Promise<PaginatedResource<Item>>)[] = []
+    pagesToFetch.forEach(page => {
+      promisesOfPagesToFetch.push(() => legacyBuilder.fetchCollectionItems(collectionId, page, limit, isReviewing))
+    })
+    const allItemPages: PaginatedResource<Item>[] = yield queue.addAll(promisesOfPagesToFetch)
+    const paginationData = allItemPages[0]
+    const items = allItemPages.map(result => result.results).flat()
+    return { items, paginationData }
+  }
+
   function* handleFetchCollectionItemsRequest(action: FetchCollectionItemsRequestAction) {
-    const { collectionId } = action.payload
+    const DEFAULT_PAGE = 1
+    const { collectionId, page = DEFAULT_PAGE, limit } = action.payload
+    const location: RouterLocation<any> = yield select(getLocation)
+    const isReviewing = location.query.reviewing === 'true'
+
     try {
-      const items: Item[] = yield call(() => legacyBuilder.fetchCollectionItems(collectionId))
-      yield put(fetchCollectionItemsSuccess(collectionId, items))
+      const { items, paginationData } = yield call(fetchCollectionItemsWithBatch, collectionId, [page], limit, isReviewing)
```

Comment:
Again I don't have enough info in my head atm but this and [this](https://github.com/decentraland/builder/pull/1921/files#diff-39d0126acb3863393af8ab7fd948f9b4ebc17e8a74bac80898b6c1d7793d9052R143) seem a bit weird. The fact that we're sending a paginationData argument which has results but then sending items separately is a bit confusing. Is results not being used from paginationData?

I am not certain if this is something that can/should be fixed

Reply:
Well, in this case, the `items` is the result of flatting an array of `Item[]`, cause it's the result of batching the requests.  It's true that `paginationData` will have items as well, but in this case it will have only the items for the first request that the queue processed. I realized that the `paginationData` was only needed for the total, so I updated the code, I think it's cleaner now :)